### DISCUSSION TITLE
Cannot make automatic package discovery work with setup.cfg when one __init__.py is missing

### DISCUSSION THREAD

**Discussion by Ark-kun** [pypa/setuptools#3719] [Q&A]
Imaging that I have a package with almost perfect "flat" structure, but with a single missing `__init__.py` due to using a submodule.

```
├── proj
│   ├── __init__.py
│   ├── _submodule
│   │   │   # No __init__.py here!
│   │   ├── tests
│   │   │  └── * many_nested_dirs
│   │   └── sub_package
│   │       ├── * many_nested_dirs
│   │       └── __init__.py
│   └── main_module
│   │       ├── * many_nested_dirs
│   │       └── __init__.py
├── setup.cfg
└── setup.py
```

I use `setup.cfg`. When I use `packages=find:`, the `sub_package` is not included in the wheel.
If I try using  `packages=find_namespace:`, many unrelated tests and test data files are included.

How can I configure the packages without having to manually include or exclude many directories.

I wish I could just do this:

```
packages=
  find: .
  find: proj/_submodule/sub_package/**
```

or

```
package_dirs=
    "": "./"
    "proj._submodule.sub_package": "proj/_submodule/sub_package/"
```

**Comment by abravalheri** [comment]
Hi @Ark-kun , I am afraid this is not possible. The best option here is to use `find_namespace` with `exclude` or `include`.
